.github/workflows/tests.yaml

```diff
@@ -31,56 +30,73 @@ jobs:
                           command: "matrix:cov",
                           matrix: "+py=3.12",
                           hatch_version: "1.0",
+                          os: ubuntu-latest,
                       }
                     - {
                           name: Python-3.11,
                           python: "3.11",
                           command: "matrix:cov",
                           matrix: "+py=3.11",
                           hatch_version: "1.0",
+                          os: ubuntu-latest,
                       }
                     - {
                           name: Python-3.10,
                           python: "3.10",
                           command: "matrix:cov",
                           matrix: "+py=3.10",
                           hatch_version: "1.0",
+                          os: ubuntu-latest,
                       }
                     - {
                           name: Python-3.9,
                           python: "3.9",
                           command: "matrix:cov",
                           matrix: "+py=3.9",
                           hatch_version: "1.0",
+                          os: ubuntu-latest,
                       }
                     - {
                           name: Python-3.8,
                           python: "3.8",
                           command: "matrix:cov",
                           matrix: "+py=3.8",
                           hatch_version: "1.0",
+                          os: ubuntu-latest,
                       }
                     - {
                           name: Hatch-1.7.x,
                           python: "3.11",
                           command: "versions:cov",
                           matrix: "+version=1.7.x",
                           hatch_version: "1.7.0",
+                          os: ubuntu-latest,
                       }
                     - {
                           name: Hatch-1.8.x,
                           python: "3.11",
                           command: "versions:cov",
                           matrix: "+version=1.8.x",
                           hatch_version: "1.8.0",
+                          os: ubuntu-latest,
                       }
                     - {
                           name: Hatch-1.9.x,
                           python: "3.11",
                           command: "versions:cov",
                           matrix: "+version=1.9.x",
                           hatch_version: "1.9.0",
+                          os: ubuntu-latest,
                       }
+                    - {
+                          name: Hatch-Windows,
+                          python: "3.11",
+                          command: "matrix:cov",
+                          matrix: "+py=3.11",
+                          hatch_version: "1.0",
+                          os: windows-latest,
+                      }
```

Comment:
This testing matrix is actually kind of problematic. See https://github.com/juftin/hatch-pip-compile/actions/runs/7450650936/job/20270034562

It looks like changing between Linux and Windows makes `hatch-pip-compile` try to recreate the lockfile.

Reply:
For some reason when running in Windows `lockfile_up_to_date` is False but `lockfile_up_to_date` is True when run on Linux.

Reply:
I split this out into #58 so this fix can unblock #56